Lib/http/server.py

```diff
@@ -769,9 +776,36 @@ def send_head(self):
                             f.close()
                             return None
 
-            self.send_response(HTTPStatus.OK)
+            if self._range:
+                start, end = self._range
+                if start is None:
+                    # `end` here means suffix length
+                    # parse_range() collapses (None, None) to None
+                    # and thus `end` can not be None here
+                    start = max(0, fs.st_size - end)
+                    end = fs.st_size - 1
+                if start >= fs.st_size:
+                    # 416 REQUESTED_RANGE_NOT_SATISFIABLE means that
+                    # none of the range values overlap the extent of
+                    # the resource
+                    f.close()
+                    headers = [('Content-Range', f'bytes */{fs.st_size}')]
+                    self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE,
+                                    extra_headers=headers)
+                    return None
+                if end is None or end >= fs.st_size:
+                    end = fs.st_size - 1
```

Comment:
Ditto about `fs.st_size = 0`

Reply:
Don’t think this can be reached for an empty file, but test cases would make this more evident

Reply:
Ah you're right. Because `start >= fs.st_size` would be satisfied (whatever `end` value we choose, whether it's allowed by the RFC or not) and we would branch on the above if and return early.